compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -4876,12 +4876,18 @@ private boolean isSameImport(BLangImportPackage importPkgNode, BPackageSymbol im
 
     private void resolveAndSetFunctionTypeFromRHSLambda(BLangVariable variable, SymbolEnv env) {
         BLangFunction function = ((BLangLambdaFunction) variable.expr).function;
-        BInvokableType invokableType = (BInvokableType) symResolver.createInvokableType(function.getParameters(),
-                                                                                        function.restParam,
-                                                                                        function.returnTypeNode,
-                                                                                        Flags.asMask(variable.flagSet),
-                                                                                        env,
-                                                                                        function.pos);
+        BType resolvedInvokableType = symResolver.createInvokableType(function.getParameters(),
+                                                                      function.restParam,
+                                                                      function.returnTypeNode,
+                                                                      Flags.asMask(variable.flagSet),
+                                                                      env,
+                                                                      function.pos);
+
+        if (resolvedInvokableType.tag == TypeTags.NONE) {
+            return;
+        }
```

Comment:
By doing this we set the type of that variable to `other`. If the type of the parameter is not defined we need to set the type of that parameter to `other`. Isn't it?
As an example 
```ballerina
var fn1 = function (astring s) returns int {
    int z = 1;
    return z;
};
```
According to the above change, we set the type of `fn1`as `other`. right? But IMO the type of `fn1` should be `function(other) returns int`
Any thoughts @MaryamZi ?

Reply:
Yeah, ideally the behaviour (errors) for both the following should be the same.

```ballerina
var fn1 = function (astring s) returns int {
    int z = 1;
    return z;
};

boolean _  = fn1("");

function fn2(astring s) returns int => 1;

boolean _  = fn2("");
```

With this change we get `undefined function 'fn1'` and don't get `incompatible types: expected 'boolean', found 'int'`.

This will require changes to `org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver#createInvokableType` though. I think we can fix this separately also since the current fix will prevent a crash.

Reply:
Will be addressed in #33836